### PR TITLE
[23.0] Serialize CWL tools via json, not pickle

### PR DESCRIPTION
**Pull request by mvdbeek**
Fixes https://github.com/galaxyproject/galaxy/actions/runs/4037394080/jobs/6940660977#step:9:7314
and reduces the diff from the cwl branch a little more.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by nsoranzo** [comment]
Can you cherry-pick https://github.com/galaxyproject/galaxy/pull/15386/commits/1fd4ce8ab0d0fd215fb66494057aa9c82c77d17e please?

Out of curiosity, do you know why this started to be a problem?

**Comment by mvdbeek** [comment]
No, I was about to dig into it, but I remembered we already did the move away from pickle. Maybe @mr-c knows ? It started a few days ago, I think. Maybe a new cwltool release ?

**Comment by mvdbeek** [comment]
I opened https://github.com/galaxyproject/galaxy/pull/15442 for the bleach fix

**Comment by mr-c** [comment]
We had a pickle problem show up in Toil recently; fix is in schema-salad >= 8.4.20230128170514

**Comment by mvdbeek** [comment]
And I think the fix is working, cause I also don't see the error in other open PRs.

**Comment by mvdbeek** [comment]
I'll retarget to dev then, there's no advantage having it in 23.0 then.

**Comment by nsoranzo** [comment]
It's also in https://github.com/galaxyproject/galaxy/actions/runs/4042381232/jobs/6950015911 , so it seems to affect unit tests on PRs targeting release_23.0 .

**Comment by mvdbeek** [comment]
Hmm, yeah, it's also in the bleach PR.

**Comment by mvdbeek** [comment]
Switched it back to 23.0, we shouldn't do pickle anyway

**Comment by nsoranzo** [comment]
Can you please remove the now unused `PERSISTED_REPRESENTATION` constant on line 86?

File: lib/galaxy/tool_util/cwl/parser.py [hunk 1]
```diff
@@ -173,29 +168,29 @@ def label(self):
...
-            raise Exception("Failed to deserialize tool proxy from JSON object - no pickle representation found.")
-        if "uuid" not in as_object:
-            raise Exception("Failed to deserialize tool proxy from JSON object - no uuid found.")
-        to_unpickle = base64.b64decode(as_object["pickle"])
-        loaded_object = pickle.loads(to_unpickle)
+        loaded_object = tool_proxy(
+            tool_object=as_object["raw_process_reference"],
+            strict_cwl_validation=strict_cwl_validation,
+            tool_directory=tool_directory,
+            uuid=as_object.get("uuid"),
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/cwl/parser.py:192]:
```suggestion
            uuid=as_object["uuid"],
```

**Review by nsoranzo — APPROVED** [review]

File: lib/galaxy/tool_util/cwl/parser.py (hunk 1 above)
Comment (mvdbeek) [reply lib/galaxy/tool_util/cwl/parser.py:192]: I am not sure that's what we wanted to do in the downstream branch. `_to_cwl_tool_object` can set the uuid, and I've spent some time in this, I'd rather not change that here.

Comment (mvdbeek) [reply lib/galaxy/tool_util/cwl/parser.py:192]: There's definitely work to be done there, I'm not sure uuid is the right thing either, this kind of replaces the tool id as a unique handle, if anything it should be the document hash, but that's also tricky due to commented yaml etc

**Pull request merged**